### DISCUSSION TITLE
How to config the supply path or parameter that can consume WIPs on different priority supply path

### DISCUSSION THREAD

**Discussion by sgfisme** [frePPLe/frepple#530] [Q&A]
Hi,

I had a supply path as below, and they had WIPs on the different paths. The netting result used the WIP1 and created a purchase order on item supplier 2. Could any possibly consume all WIPs on the entire supply path first?

![image](https://user-images.githubusercontent.com/44051141/172409151-7aaba2f9-f67f-4777-ae00-c80131608888.png)

**Comment by jdetaeye** [comment]
This is only possible with a customization.   

You'll need a first planning sweep during which all procurement is disabled (eg by setting the priority of the itemsuppliers to 0). The planning algorithm is then forced to search for WIP on all alternates.
In a second planning sweep you then plan the remaining quantities that can't be supplied from WIP.

**Reply by sgfisme** [reply]
Thanks. @jdetaeye 
I had the other scenario that 1 WIP was on the main route and executed the plan. Finally, the plan didn't choose the WIP and generated two purchase orders on the item supplier 3 & 4 to fulfill an order directly.

I expected the plan would pick the WIP and the remaining order quantities would generate the purchase order to fulfill. Is this only possible with customization for this scenario?

![image](https://user-images.githubusercontent.com/44051141/173259677-4dec7d62-ac82-443b-aae5-8c9ef60d8cd7.png)

**Comment by jdetaeye** [comment]
If 1) the operation "search mode" is "priority" and 2) the WIP1 is available within the window specified in the "plan.autofence" parameter, then the WIP should have been used and a procurement for the remaining quantity is generated. 
Looks like these conditions aren't met in your model.

If the due demand of the demand is very near, and the lead time on the priority 2 branch is shorter than the priority 1 branch of the network, then it's perfectly fine to use the priority 2 branch.

To understand in detail you need to run with the parameter plan.loglevel = 2 and follow how the algorithms searches the supply path.

**Reply by sgfisme** [reply]
Thanks, @jdetaeye.  
I checked the “plan.autoFenceOperations” parameter and we set this value as 0. This is probably why our plan didn’t consume the WIP1.

If I hoped that WIP1 had been consumed, but don’t wait a lot of time for generating purchase orders. How do assess this parameter setting?

**Reply by jdetaeye** [reply]
There is an example model on this feature: https://frepple.org/docs/current/user-guide/examples/operation/operation-autofence.php   
This picture https://frepple.org/docs/current/user-guide/_images/autofence.png from there also explains it well.

The default value is 0 = "never wait" for "historical reasons" & backward compatibility.    I recommend setting it to 999 = "always wait", unless your business has the (questionable & bad!) practice of ordering way too early (eg ordering a 7-day-lead-time item now for delivery next year).
